crates/integration-tests/src/rspamd.rs

```diff
@@ -0,0 +1,413 @@
+                .and_then(|h| h.as_unstructured().ok()),
+            spam_action: headers
+                .get_first("X-Spam-Action")
+                .and_then(|h| h.as_unstructured().ok()),
+        })
+    }
+
+    /// Assert that X-Spam-Flag header is present
+    fn assert_has_flag(&self) -> anyhow::Result<&Self> {
+        anyhow::ensure!(
+            self.spam_flag.is_some(),
+            "Expected X-Spam-Flag header to be present"
+        );
+        Ok(self)
+    }
+
+    /// Assert that X-Spam-Flag header has a specific value
+    fn assert_flag_equals(&self, expected: &str) -> anyhow::Result<&Self> {
+        match &self.spam_flag {
+            Some(flag) => {
+                anyhow::ensure!(
+                    flag == expected,
+                    "Expected X-Spam-Flag to be '{}', got '{}'",
+                    expected,
+                    flag
+                );
+            }
+            None => {
+                anyhow::bail!("Expected X-Spam-Flag header to be present");
+            }
+        }
+        Ok(self)
+    }
+
+    /// Assert that X-Spam-Score header is present and can be parsed as a float
+    fn assert_has_score(&self) -> anyhow::Result<f64> {
+        match &self.spam_score {
+            Some(score) => {
+                let parsed = score.parse::<f64>().context(format!(
+                    "Failed to parse X-Spam-Score '{}' as float",
+                    score
+                ))?;
+                Ok(parsed)
+            }
+            None => {
+                anyhow::bail!("Expected X-Spam-Score header to be present");
+            }
+        }
+    }
+
+    /// Assert that X-Spam-Action header is present
+    fn assert_has_action(&self) -> anyhow::Result<&Self> {
+        anyhow::ensure!(
+            self.spam_action.is_some(),
+            "Expected X-Spam-Action header to be present"
+        );
+        Ok(self)
+    }
+
+    /// Assert that X-Spam-Action header has a specific value
+    fn assert_action_equals(&self, expected: &str) -> anyhow::Result<&Self> {
+        match &self.spam_action {
+            Some(action) => {
+                anyhow::ensure!(
+                    action == expected,
+                    "Expected X-Spam-Action to be '{}', got '{}'",
+                    expected,
+                    action
+                );
+            }
+            None => {
+                anyhow::bail!("Expected X-Spam-Action header to be present");
+            }
+        }
+        Ok(self)
+    }
+}
+
+#[tokio::test]
+async fn test_rspamd_scan_message() -> anyhow::Result<()> {
+    if std::env::var("KUMOD_TESTCONTAINERS").unwrap_or_else(|_| String::new()) != "1" {
+        return Ok(());
+    }
+
+    let rspamd = RspamdContainer::new().await?;
+
+    let policy_file = std::path::PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("rspamd.lua");
+
+    let mut daemon = crate::kumod::DaemonWithMaildirOptions::new()
+        .policy_file(policy_file.to_string_lossy().to_string())
```

Comment:
this is more consistent with how the other tests do this:

```suggestion
        .policy_file("rspamd.lua")
```